import-automation/workflow/helper/spanner_client.py

```diff
@@ -0,0 +1,158 @@
+            params = {"lockId": self._LOCK_ID}
+            param_types = {"lockId": STRING}
+
+            current_owner = None
+            results = transaction.execute_sql(sql, params, param_types)
+            for row in results:
+                current_owner = row[0]
+
+            if current_owner == workflow_id:
+                sql = """
+                    UPDATE IngestionLock
+                    SET LockOwner = NULL, AcquiredTimestamp = NULL
+                    WHERE LockID = @lockId
+                """
+                transaction.execute_update(sql,
+                                           params={"lockId": self._LOCK_ID},
+                                           param_types={"lockId": STRING})
+                logging.info(f"Lock successfully released by {workflow_id}")
+                return True
+            else:
+                logging.info(f"Lock is currently held by {current_owner}")
+                return False
+
+        return self.database.run_in_transaction(_release)
+
+    def get_import_status(self):
+        """Get the list of pending imports to ingest."""
+        pending_imports = []
+        sql = "SELECT ImportName, Latestversion FROM ImportStatus WHERE State = 'PENDING'"
+        # Use a read-only snapshot for this query
+        with self.database.snapshot() as snapshot:
+            results = snapshot.execute_sql(sql)
+            for row in results:
+                import_json = {}
+                import_json['importName'] = row[0]
+                import_json['latestVersion'] = row[1]
+                pending_imports.append(import_json)
+
+        logging.info(f"Found {len(pending_imports)} import jobs as PENDING.")
+        return pending_imports
+
+    def update_import_status(self, workflow_id: str, import_list: list) -> None:
+        """Atomically marks pending imports as DONE and records the ingestion event."""
+        succeeded_imports = []
+        for import_json in import_list:
+            succeeded_imports.append(import_json['importName'])
+
+        def _record(transaction: Transaction) -> None:
+            # 1. Update the ImportStatus table
+            update_sql = "UPDATE ImportStatus SET State = 'DONE', JobId = @workflowId, UpdateTimestamp = PENDING_COMMIT_TIMESTAMP() WHERE ImportName IN UNNEST(@importNames)"
+            updated_rows = transaction.execute_update(
+                update_sql,
+                params={
+                    "importNames": succeeded_imports,
+                    "workflowId": workflow_id
+                },
+                param_types={
+                    "importNames": Array(STRING),
+                    "workflowId": STRING
+                })
+            logging.info(f"Marked {updated_rows} import jobs as DONE.")
+
+            # 2. Insert into the IngestionHistory table
+            insert_sql = """
+                INSERT INTO IngestionHistory (CompletionTimestamp, WorkflowExecutionID, IngestedImports)
```

Comment:
To confirm, are the recent rows with empty IngestedImports failed dataflow jobs?